### PR TITLE
Add all units section on docs

### PR DESCRIPTION
**Pull request by williamjamir**
Link for the docs

https://barril.readthedocs.io/en/add-all-units/

The section `List of physical units` was added,

image

CC @tadeu @arthursoprana @igortg @tigarmo @GabrielMarcilio

### CODE REVIEW

File: docs/_extension/list_all_units.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+from docutils import nodes
+from docutils.statemachine import ViewList
+from sphinx.util import nested_parse_with_titles
+from sphinx.util.docutils import SphinxDirective
+
+
+class ListAllUnits(SphinxDirective):
+    def run(self):
+        rst = ViewList()
+        unit_database = Scalar(10, "m").GetUnitDatabase()
```
Comment (nicoddemus) [inline docs/_extension/list_all_units.py]:
```suggestion
        unit_database = UnitDatabase.GetSingleton()
```

File: docs/_extension/list_all_units.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        source = "fakefile.rst"
+
+        # Create the rst content
+        for categories in sorted(unit_database.IterCategories(), key=str.casefold):
```
Comment (nicoddemus) [inline docs/_extension/list_all_units.py]: 👍 on `casefold`

Comment (nicoddemus) [inline docs/_extension/list_all_units.py]:
```suggestion
        for category in sorted(unit_database.IterCategories(), key=str.casefold):
```

File: docs/_extension/list_all_units.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            title = unit_database.GetCategoryInfo(categories).caption
```
Comment (nicoddemus) [inline docs/_extension/list_all_units.py]:
```suggestion
            title = unit_database.GetCategoryInfo(categories).caption.title()
```

File: docs/_extension/list_all_units.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+            rst.append(f".. rubric:: {title}", source)
+            rst.append(f"", source)
+
+            for unit in unit_database.GetCategoryInfo(categories).valid_units_set:
+                unit = unit_database.unit_to_unit_info[unit].unit
+                name = unit_database.unit_to_unit_info[unit].name
```
Comment (nicoddemus) [inline docs/_extension/list_all_units.py]: * This should be sorted by `unit`
* Why `unit = unit_database.unit_to_unit_info[unit].unit`?

**Comment by nicoddemus** [comment]
Please also build the docs (with `-W`) during `linting`).

File: docs/_extension/list_all_units.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
+from barril.units import Scalar
+from docutils import nodes
+from docutils.statemachine import ViewList
+from sphinx.util import nested_parse_with_titles
+from sphinx.util.docutils import SphinxDirective
+
```
Comment (williamjamir) [inline docs/_extension/list_all_units.py]:
```suggestion
from barril.units import UnitDatabase

File: docs/_extension/list_all_units.py (hunk 4 above)
Comment (williamjamir) [reply docs/_extension/list_all_units.py]: > This should be sorted by unit
Good call =)

>> Why `unit = unit_database.unit_to_unit_info[unit].unit?`
🤦‍♂

Comment (williamjamir) [reply docs/_extension/list_all_units.py]: Commit 91515c56de1862d4efc11b7dc086f11664c45c93 fix this issue

**Comment by williamjamir** [comment]
> Please also build the docs (with `-W`) during `linting`).

Commit e8584ab432fc44b006658cb4605924ff23a8f031 invoke the docs toxenv during `linting`job

File: docs/_extension/list_all_units.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        for category in sorted(unit_database.IterCategories(), key=str.casefold):
+            title = unit_database.GetCategoryInfo(category).caption.title()
+            rst.append(f".. rubric:: {title}", source)
+            rst.append(f"", source)
+
+            for unit in sorted(
+                unit_database.GetCategoryInfo(category).valid_units_set, key=str.casefold
+            ):
+                name = unit_database.unit_to_unit_info[unit].name
+                rst.append(f'- "{unit}" ({name})', source)
```
Comment (nicoddemus) [inline docs/_extension/list_all_units.py]:
```suggestion
                rst.append(f'- ``"{unit}"`` ({name})', source)
```

File: docs/units.rst [hunk 7]
```diff
@@ -0,0 +1,4 @@
+List of physical units
```
Comment (nicoddemus) [inline docs/units.rst]:
```suggestion
List of Physical Units
```

Comment (nicoddemus) [reply docs/units.rst]: image

**Comment by tadeu** [comment]
awesome addition :D

**Pull request merged**